src/main/java/Problem2/SingleLinkedList.java

```diff
@@ -13,18 +13,48 @@ public SingleLinkedList() {
     // copy constructor
     public SingleLinkedList(SingleLinkedList list) {
         // homework
+        this.head = list.head;
```

Comment:
wrong implementation of copy constructor.

this is called out in description for this problems

```
Note that copy constructor will copy the values in each node, not the references.
```